tests/starlite/schema.py

```diff
@@ -134,7 +134,7 @@ async def flavors(self) -> typing.AsyncGenerator[Flavor, None]:
     @strawberry.subscription
     async def debug(self, info) -> typing.AsyncGenerator[DebugInfo, None]:
         active_result_handlers = [
-            task for task in info.context["tasks"].values() if not task.done()
+            task for task in info.context["get_tasks"]() if not task.done()
```

Comment:
Interesting, could you elaborate on putting a get_tasks method instead of the tasks list in the context? My first thought was that it would make sure that the tasks are always up to date when the debug subscription is used. Is that the intention? In that case I'm wondering (I'm not sure asking to make sure) whether the list would always be updated even without that change since it's never copied but passed by reference 🤔

Reply:
So, yes:  Previously, both handler classes had a `tasks` member which was just a `dict`.  A reference to this dict was in the context.  And a fresh view of the active tasks could be got at any point by executing its `values()` member.

With this change, the tasks are not immediately available like that.  It also, imho, is too coupled to the implementation, even for testing code.  So instead, we have a "getter" method in the context.

The change to the tests isn't really that big, instead of having to call  `values()` on the property, it just needs to _call_ the property.

Reply:
Thanks for clearifying. So to summarize: The idea behind the `get_tasks` getting is to abstract the internals of task handling away from the test code shared by the different integrations. (correct? :eyes: )

Reply:
Hi, sorry for late answer.  It's that, yes, and also so the the test code for the individual integrations need not know exactly how the handlers are storing their asyncio tasks, and if they do so differently.  It is enough to call the special get_tasks() method on the context created by the debuggable schema.

Reply:
Thanks for clearifying!